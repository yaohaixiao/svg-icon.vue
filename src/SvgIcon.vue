<template>
  <svg
    v-if="name"
    aria-hidden="true"
    :style="cssRules"
    class="svg-icon">
    <use v-bind="binds" />
  </svg>
</template>

<script>
/**
 * SvgIcon.vue - SVG 图标显示组件
 * =============================================================
 * Created By: Yaohaixiao
 * Update: 2022.10.8
 */
import { render } from './utils/utils'

render()

export default {
  name: 'SvgIcon',
  componentName: 'SvgIcon',
  props: {
    name: {
      type: String,
      default: ''
    },
    size: {
      type: [
        String,
        Number
      ],
      default: 16
    },
    color: {
      type: String,
      default: ''
    }
  },
  computed: {
    binds() {
      return {
        'xlink:href': `#icon-${this.name}`
      }
    },
    cssRules() {
      const size = this.size
      const color = this.color

      return {
        width: `${size}px`,
        height: `${size}px`,
        color: color
      }
    }
  }
}
</script>

<style lang="less">
@import './svg-icon';
</style>
