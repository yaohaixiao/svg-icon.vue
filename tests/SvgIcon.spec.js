import {
  shallowMount
} from '@vue/test-utils'

import SvgIcon from '../src/SvgIcon'

describe('SvgIcon.vue', () => {
  let wrapper
  const DEFAULT_SIZE = 16
  const DEFAULT_NAME = 'edit'
  const DEFAULT_COLOR = '#DDD'

  beforeAll(() => {
    wrapper = shallowMount(SvgIcon, {
      // 配置组件 props 属性
      propsData: {
        size: DEFAULT_SIZE,
        name: DEFAULT_NAME,
        color: DEFAULT_COLOR
      }
    })
  })

  it('生成组件渲染快照', () => {
    expect(wrapper.element).toMatchSnapshot()
  })

  it(`props.name 默认值：${DEFAULT_NAME}`, () => {
    expect(wrapper.vm.name).toEqual(DEFAULT_NAME)
  })

  it(`props.size 默认值：${DEFAULT_SIZE}`, () => {
    expect(wrapper.vm.size).toEqual(DEFAULT_SIZE)
  })

  it(`props.color 默认值：${DEFAULT_COLOR}`, () => {
    expect(wrapper.vm.color).toEqual(DEFAULT_COLOR)
  })

  it(`computed.binds 值：{'xlink:href': #icon-${DEFAULT_NAME}}`, () => {
    const binds = {
      'xlink:href': `#icon-${wrapper.vm.name}`
    }
    expect(wrapper.vm.binds).toEqual(binds)
  })

  it('调整 props.name 值为：cancel', async () => {
    const CANCEL = 'cancel'
    wrapper.setProps({
      name: CANCEL
    })
    await wrapper.vm.$nextTick()
    const binds = {
      'xlink:href': `#icon-${wrapper.vm.name}`
    }
    expect(wrapper.vm.name).toEqual(CANCEL)
    expect(wrapper.vm.binds).toEqual(binds)
  })

  it('调整 props.size 值为：28', async () => {
    const SIZE = 28
    const CANCEL = 'cancel'
    wrapper.setProps({
      size: SIZE,
      name: CANCEL
    })
    await wrapper.vm.$nextTick()
    const binds = {
      'xlink:href': `#icon-${wrapper.vm.name}`
    }
    expect(wrapper.vm.size).toEqual(SIZE)
    expect(wrapper.vm.binds).toEqual(binds)
  })

  it('调整 props.color 值为：#f00', async () => {
    const COLOR = '#f00'
    const CANCEL = 'cancel'
    wrapper.setProps({
      color: COLOR,
      name: CANCEL
    })
    await wrapper.vm.$nextTick()
    const binds = {
      'xlink:href': `#icon-${wrapper.vm.name}`
    }
    expect(wrapper.vm.color).toEqual(COLOR)
    expect(wrapper.vm.binds).toEqual(binds)
  })
})
