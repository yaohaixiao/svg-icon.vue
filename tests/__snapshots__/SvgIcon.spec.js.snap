// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SvgIcon.vue 生成组件渲染快照 1`] = `
<svg
  aria-hidden="true"
  class="svg-icon"
  style="width: 16px; height: 16px; color: rgb(221, 221, 221);"
>
  <use
    xlink:href="#icon-edit"
  />
</svg>
`;
